src/unix/thread.c

```diff
@@ -204,6 +205,77 @@ int uv_thread_create(uv_thread_t *tid, void (*entry)(void *arg), void *arg) {
 }
 
 
+#if defined(__APPLE__) && defined(__MACH__) || defined(_AIX)
+int uv_thread_setaffinity(uv_thread_t* tid,
+                          char* cpumask,
+                          char* oldmask,
+                          size_t mask_size) {
+  return -ENOTSUP;
+}
+
+
+int uv_thread_getaffinity(uv_thread_t* tid,
+                          char* cpumask,
+                          size_t mask_size) {
+  return -ENOTSUP;
+}
+
+#else /* !((defined(__APPLE__) && defined(__MACH__)) || defined(_AIX)) */
+
+int uv_thread_setaffinity(uv_thread_t* tid,
+                          char* cpumask,
+                          char* oldmask,
+                          size_t mask_size) {
+  int i;
+  int r;
+  cpu_set_t cpuset;
+  int cpumasksize;
+
+  cpumasksize = uv_cpumask_size();
+  assert(mask_size >= (size_t)cpumasksize);
```

Comment:
Does the `mask_size` argument refer to the size of `oldmask`?. If that's the case why don't leave the assertion/check to `uv_thread_getaffinity()`?